### PR TITLE
feat(logging): add client side logs

### LINKED ISSUE #2727: Add client-side logging
Would help in investigating users' issues that haven't given permission for us to collect remote diagnostics.

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/3319312312).

We'll benefit from having logs we can ask users to share with us, when encountering problems with the wallet. This PR adds an append-only logs store. 

Rxjs is used to queue appends, making sure one write action doesn't overwrite another.

In the below demo, I first generate a handful of test logs, copy them to clipboard, then delete them. Holding <kbd>alt</kbd> reveals the advanced diagnostics menu.

https://user-images.githubusercontent.com/1618764/196733898-42487e25-378b-42d0-9a83-e66ef536ff32.mp4

### CODE REVIEW

File: src/app/common/transactions/broadcast-transaction.ts [hunk 1]
```diff
@@ -30,14 +30,16 @@ export async function broadcastTransaction(options: BroadcastTransactionOptions)
   );
 
   if ('error' in response) {
-    logger.error(`Error broadcasting raw transaction -- ${response}`);
+    logger.error(`Error broadcasting raw transaction`, response);
     const message = getErrorMessage(response.reason as any);
     throw new Error(`${message} - ${(response as any).error}`);
   } else if (!validateTxId(response.txid)) {
-    logger.error(`Error broadcasting raw transaction -- ${response}`);
+    logger.error(`Error broadcasting raw transaction, response`);
```
Comment (fbwoolf) [inline src/app/common/transactions/broadcast-transaction.ts]:
```suggestion
    logger.error(`Error broadcasting raw transaction`, response);
```

**Review by fbwoolf — APPROVED** [review]
Looks great, this will be really helpful! 👍

File: src/shared/logger.ts [hunk 2]
```diff
@@ -1,15 +1,51 @@
...
 
-export const logger = {
-  info: loggingTool.info,
-  warn: loggingTool.warn,
-  debug: loggingTool.debug,
-  error: loggingTool.error,
+type LoggerFnApi = (msg: string, ...args: any[]) => void;
+
+// Pino offers a handful of APIs to log messages. Here we enforce a single one,
+// decribed by `LoggerFnApi`
```
Comment (edu-stx) [inline src/shared/logger.ts]:
```suggestion
// described by `LoggerFnApi`
```

**Review by edu-stx — APPROVED** [review]
looking good

**Pull request merged**